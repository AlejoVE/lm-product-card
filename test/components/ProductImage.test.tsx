import React from 'react';

import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src';
import { product } from '../data/products';

describe('ProductImage', () => {
  test('should show title correctly with custom title', () => {
    const wrapper = renderer.create(<ProductImage img="https://hola.org" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show component with the image of the product', () => {
    const wrapper = renderer.create(
      <ProductCard product={product}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
