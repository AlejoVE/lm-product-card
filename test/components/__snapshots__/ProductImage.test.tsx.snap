// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should show component with the image of the product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage should show title correctly with custom title 1`] = `
<img
  alt="Coffee Mug"
  className="productImg undefined"
  src="https://hola.org"
/>
`;
